src/legacy/core_plugins/data/public/search/aggs/buckets/lib/date_range.ts

```diff
@@ -17,20 +17,64 @@
  * under the License.
  */
 
+import dateMath from '@elastic/datemath';
+import { TimeBuckets } from './time_buckets';
+import { TimeRange } from '../../../../../../../../plugins/data/public';
+import {
+  getFieldFormats,
+  getUiSettings,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../../../../../../plugins/data/public/services';
+
 export interface DateRangeKey {
   from: number;
   to: number;
 }
 
-export const convertDateRangeToString = (
-  { from, to }: DateRangeKey,
-  format: (val: any) => string
-) => {
+export function convertDateRangeToString({ from, to }: DateRangeKey, format: (val: any) => string) {
   if (!from) {
     return 'Before ' + format(to);
   } else if (!to) {
     return 'After ' + format(from);
   } else {
     return format(from) + ' to ' + format(to);
   }
-};
+}
+
+export function toAbsoluteDates(range?: TimeRange) {
+  if (!range) {
+    return;
+  }
+
+  const fromDate = dateMath.parse(range.from);
+  const toDate = dateMath.parse(range.to, { roundUp: true });
+
+  if (!fromDate || !toDate) {
+    return;
+  }
+
+  return {
+    from: fromDate.toDate(),
+    to: toDate.toDate(),
+  };
+}
+
+export function getAutoIntervalFromTimeRange(range?: TimeRange, defaultValue: string = '1h') {
```

Comment:
this probably doesn't belong on date_range but rather on date_histogram or just separate utilities file